src/membership/schemas/membership.schema.ts

```diff
@@ -0,0 +1,47 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+
+@Schema({ collection: 'memberships' })
+export class Membership extends Document {
+  @Prop({ required: true, minlength: 5, maxlength: 50 })
+  name: string;
+
+  @Prop({ required: true, unique: true })
+  mobileNo: string;
+
+  @Prop({ required: true, unique: true })
+  email: string;
+
+  @Prop({ required: true })
+  currentCountry: string;
+
+  @Prop()
+  profession: string;
+
+  @Prop()
+  professionalInstitute: string;
+
+  @Prop()
+  designation: string;
+
+  @Prop()
+  lastAcademicLevel: string;
+
+  @Prop()
+  lastPassingYear: number;
+
+  @Prop()
+  lastBatch: string;
+
+  @Prop({ required: true })
+  paymentTransactionId: string;
+
+  @Prop({
+    required: true,
+    enum: ['pending', 'approved', 'rejected'],
```

Comment:
Use actual enum to assign these values

Create an enum like that 

```typescript
export enum MembershipStatus {
    Pending = 'pending',
    Approved = 'approved',
    Rejected = 'rejected'
}
```

Then use the enum in the prop enum

```suggestion
    enum: MembershipStatus,
    default: MembershipStatus.Pending
```

Additionally you can check if the value is actually enum or not by - 

```suggestion
    @Prop({
        type: String,
        enum: MembershipStatus,
        default: MembershipStatus.Pending
    })
    @IsEnum(MembershipStatus)
    status: string;
```